test/groovy/BuildDockerAndPublishImageStepTests.groovy

```diff
@@ -117,9 +117,13 @@ class BuildDockerAndPublishImageStepTests extends BaseTest {
   // Return if the usual static checks had been recorded with the usual pattern
   Boolean assertRecordIssues(String imageName = fullTestImageName) {
     final String reportId = "${imageName}-hadolint-${mockedTimestamp}".replaceAll('/','-').replaceAll(':', '-')
+    boolean skipChecks = false
+    if (env.BRANCH_IS_PRIMARY) {
```

Comment:
you tested this? as an env variable will be a string this _may_ not work as expected

Reply:
Yep, I've updated the PR body with the tests after your comment.

The issue is that, due to the pipeline engine, the env.BRANCH_IS_PRIMARY was undefined (`null`) when running in a PR (hence the error message we had). No type inference was done so the `recordIssue()` function, expecting a boolean value for the argument `skipPublishingSteps` receieved a `null` value.

With this new code, using the Groovy features, https://groovy-lang.org/semantics.html#_strings empty string is coerced to `false` for the `if` condition and to `true` if not empty.

It's a codestyle tradeoff: it adds a code step, but it avoid letting the pipeline engine coercing types in the step call

Reply:
Groovy languages recommends to rely the emptiness coercion https://groovy-lang.org/semantics.html#_strings => it is a feature language we can rely on when using pure groovy instruction.

The inference issue came from calling Groovy feature in a pipeline step which can have.. weird behavior (hence the choice to move it in the  `if` case


Your suggestion is valid but un-needed. Would be true on Javascript though \o/